spec/models/person_escort_record_spec.rb

```diff
@@ -216,4 +216,65 @@
       expect(FrameworkResponse.count).to be_zero
     end
   end
+
+  describe '#section_progress' do
+    it 'returns an empty hash if no responses present' do
+      person_escort_record = create(:person_escort_record)
+
+      expect(person_escort_record.section_progress).to be_empty
+    end
+
+    it 'returns all section values for framework questions' do
+      person_escort_record = create(:person_escort_record, :with_responses)
+      question_sections = person_escort_record.framework_questions.pluck(:section).uniq
+      progress_sections = person_escort_record.section_progress.map { |section| section[:key] }
+
+      expect(progress_sections).to contain_exactly(*question_sections)
```

Comment:
You could use `match_array(question_sections)` here instead of splatting the argument - I think